### ISSUE TITLE
The thread too much

### ISSUE THREAD

**Issue by gaoguofan** [alibaba/nacos#8964]
**Describe the bug**
When using dubbo to integrate with nacos, there are a lot of different types of threads found.

**Expected behavior**
According to common sense, the role of this part of the thread pool is inferred. The first is to publish the heartbeat of the interface, and the other is to obtain interface data through long polling.But the number of these two threads should not be too much, a single thread is enough to traverse the status of all interfaces and the business of heartbeat

**Actually behavior**
A clear and concise description of what you actually to happen.
The version of the nacos client I am using is 1.2, and the names of these two threads are “com.alibaba.nacos.client.naming.updater” and com.alibaba.nacos.naming.beat.sender

**Comment by WilliamsSj** [comment]
We confirm that the bug exists and we plan to fix it in the advanced 3.0 release.

**Comment by KomachiSion** [comment]
you can update to 2.0 version, in 2.0 version the heartbeat is connection level, not interface level.

**Comment by KomachiSion** [comment]
and you can see whether you use dubbo 2.7.8, which has bugs to create quick many threads.

**Comment by ijustyce** [comment]
> 

can you give me some link about this question please.

**Comment by gaoguofan** [comment]
> > 
> 
> can you give me some link about this question please.
No link yet, but I can give you a description of the scenario I use.
I used a self-developed gateway and needed to consume multiple dubbo interfaces. At this time, there will be many threads related to nacos, such as idle, sender threads, etc.
I found through source code analysis that when each dubbo interface generates an agent, a thread pool of 8 threads will be created. Note that every dubbo interface will do this, but I found through source code analysis that the relevant parameters are configured in registryConfig , the number of threads in the generated thread pool can be stabilized at 1, and the configuration parameters are as follows:
<img width="759" alt="image" src="https://user-images.githubusercontent.com/12770622/184819776-71fcfa09-2f44-4026-8f01-e290b894586f.png">

**Comment by gaoguofan** [comment]
No link yet, but I can give you a description of the scenario I use.
I used a self-developed gateway and needed to consume multiple dubbo interfaces. At this time, there will be many threads related to nacos, such as idle, sender threads, etc.
I found through source code analysis that when each dubbo interface generates an agent, a thread pool of 8 threads will be created. Note that every dubbo interface will do this, but I found through source code analysis that the relevant parameters are configured in registryConfig , the number of threads in the generated thread pool can be stabilized at 1, and the configuration parameters are as follows:
<img width="719" alt="image" src="https://user-images.githubusercontent.com/12770622/184819976-43363925-2b6e-40ac-835b-0f5104b98d3b.png">
